### PR TITLE
feat(specs): add composition API

### PR DESCRIPTION
**Pull request by e-krebs**
## 🧭 What and Why

🎟 JIRA Ticket:

### Changes included:

this pr adds a new package for the javascript client, for the composition API. we only provide an alpha version, in javascript for now, and will make it globally available (and in algoliasearch) once the stable version has been reached.

### CODE REVIEW

**Comment by shortcuts** [comment]
good @e-krebs everything should work once our testing app has been allowed to use composition, so that our e2e can runs

**Comment by e-krebs** [comment]
> good @e-krebs everything should work once our testing app has been allowed to use composition, so that our e2e can runs

is it a metis App ?

**Comment by shortcuts** [comment]
> > good @e-krebs everything should work once our testing app has been allowed to use composition, so that our e2e can runs
> 
> is it a metis App ?

nope, is it required? we could migrate it or make a new one

**Comment by e-krebs** [comment]
> > > good @e-krebs everything should work once our testing app has been allowed to use composition, so that our e2e can runs
> > 
> > is it a metis App ?
> 
> nope, is it required? we could migrate it or make a new one

yes, it is required.
once you have a metis app it's fairly easy to activate composition

**Comment by shortcuts** [comment]
> > > > good @e-krebs everything should work once our testing app has been allowed to use composition, so that our e2e can runs
> > > 
> > > 
> > > is it a metis App ?
> > 
> > 
> > nope, is it required? we could migrate it or make a new one
> 
> yes, it is required. once you have a metis app it's fairly easy to activate composition

ok! i'll be off next week so I guess @millotp will be the point of contact to help setting those for the e2e

File: specs/composition/common/schemas/compositionBehavior.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+              title: compositionSource
+              type: object
+              additionalProperties: false
+              properties:
+                search:
+                  title: compositionSourceSearch
+                  type: object
+                  additionalProperties: false
+                  properties:
+                    index:
```
Comment (millotp) [inline specs/composition/common/schemas/compositionBehavior.yml:25]: In the search API and all the other spec, the property for an index name is always called `indexName` and not just `index`.
This is just a matter of consistency, I saw that in the composition spec `indexName` is also used

Comment (e-krebs) [reply specs/composition/common/schemas/compositionBehavior.yml:25]: I agree in theory, but this property is named `index` in the API itself, according to the documentation: https://algolia.atlassian.net/wiki/spaces/CMP/pages/5149753345/Models

**Review by millotp — APPROVED** [review]
Thanks for the contribution, let release a first version !

**Pull request merged**